### PR TITLE
ESQL: Fix logging test

### LINKED ISSUE #108367: [CI] RestEsqlIT testDoLogWithDebug {p0=SYNC} failing on serverless
```
./gradlew ':qa:stateful:x-pack:plugin:esql:qa:server:single-node:javaRestTest' --tests "org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT.testDoLogWithDebug {p0=SYNC}" -Dtests.seed=5B90295C80044DC9 -Dtests.locale=sk -Dtests.timezone=America/Lower_Princes -Druntime.java=21
```

```
java.lang.AssertionError: 
Expected: <true>
     but: was <false>

  at __randomizedtesting.SeedInfo.seed([5B90295C80044DC9:19863785B9E9644]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
  at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2160)
  at org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT.testDoLogWithDebug(RestEsqlIT.java:145)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)
```

### PR DESCRIPTION
**Pull request by nik9000**
This fixes the logging tests when run against multiple nodes.

Closes #108367

### CODE REVIEW

File: x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java [hunk 1]
```diff
@@ -136,14 +137,16 @@ public void testDoLogWithDebug() throws IOException {
...
-                boolean[] found = new boolean[] { false };
-                Streams.readAllLines(log, line -> {
-                    if (line.contains("DO_LOG_ME")) {
-                        found[0] = true;
-                    }
-                });
-                assertThat(found[0], equalTo(true));
+            boolean[] found = new boolean[] { false };
+            for (int i = 0; i < cluster.getNumNodes(); i++) {
+                try (InputStream log = cluster.getNodeLog(0, LogType.SERVER)) {
```
Comment (luigidellaquila) [inline x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:142]: I guess the intention here was to check logs on each node

```suggestion
                try (InputStream log = cluster.getNodeLog(i, LogType.SERVER)) {
```

File: x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java [hunk 2]
```diff
@@ -136,14 +137,16 @@ public void testDoLogWithDebug() throws IOException {
...
+                    Streams.readAllLines(log, line -> {
+                        if (line.contains("DO_LOG_ME")) {
+                            found[0] = true;
+                        }
+                    });
+                    assertThat(found[0], equalTo(true));
+                }
             }
```
Comment (luigidellaquila) [inline x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:150]: Do we expect logs on all the nodes? Shouldn't the assertion be after the for loop?
```suggestion
                }
            }
            assertThat(found[0], equalTo(true));
```

**Review by luigidellaquila — COMMENTED** [review]
If I got the intention right, it needs a little fix; apart from that, it looks good.
Thanks!

File: x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java (hunk 2 above)
Comment (bpintea) [reply x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:150]: > Do we expect logs on all the nodes? Shouldn't the assertion be after the for loop? 

The `test-update-serverless` label should have found this.

Comment (nik9000) [reply x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:150]: For science!

**Comment by nik9000** [comment]
> If I got the intention right, it needs a little fix; apart from that, it looks good.

oops. Those all make sense. I'll get them in. First I'll make sure the serverless build fails without them. I mean, it should, but if it doesn't I'll be confused.

**Review by luigidellaquila — APPROVED** [review]

**Pull request merged**